tests/bluetooth/bsim_bt/bsim_test_rpa_peripheral/src/dut.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bs_bt_utils.h"
+#include "zephyr/bluetooth/addr.h"
+#include "zephyr/bluetooth/bluetooth.h"
+#include "zephyr/bluetooth/conn.h"
+#include "zephyr/toolchain/gcc.h"
+#include <host/hci_core.h>
+
+#include <stdint.h>
+#include <string.h>
+
+static struct bt_le_ext_adv_start_param start_params;
+static struct bt_le_ext_adv_cb cb_adv;
+static struct bt_le_ext_adv *adv;
```

Comment:
Can be moved to `start_advertising`

Reply:
Test will fail if these declarations are moved to `start_advertising` (`cb_rpa_expired` will never get triggered).

Interestingly, if these declarations and the contents of `start_advertising` are all moved to `dut_procedure`, this problem does not occur.
I am uncertain why.

Reply:
`cb_adv` will need to be `static` - Is that perhaps the issue?

Reply:
Yes, it was. Thanks !